RecoLocalTracker/SiStripZeroSuppression/src/FastLinearCMNSubtractor.cc

```diff
@@ -13,7 +13,7 @@ inline void FastLinearCMNSubtractor::subtract_(uint32_t detId, uint16_t firstAPV
   tmp.reserve(128);
   typename std::vector<T>::iterator strip(digis.begin()), end(digis.end()), endAPV, high, low;
 
-  while (strip < end) {
+  while (strip - end < 0) {
```

Comment:
looking at https://en.cppreference.com/w/cpp/named_req/RandomAccessIterator
<img width="396" alt="image" src="https://user-images.githubusercontent.com/4676718/178279530-f74fa3e9-455a-4a9a-b80d-af47c999c4b4.png">

why would a semantically equivalent expression trigger a warning?

Reply:
Hi,

After some further investigation, I think in this case the compiler is warning about a possible pointer wraparound since, for example, when checking `high < endAPV`, it is in reality checking for:

```
endAPV = strip + 128;
high = strip + 64;
```

My fear about `high - endAPV < 0` is that the warning goes away, but probably also the optimization and that is why the warning is not shown anymore.

Reply:
this is what I understood reading about similar reports from 2014

Reply:
Hi @makortel,

Many thanks for your answer. I was also thinking on the option of disabling the warnings only on the specific code locations where it appears. I have been doing some trials with `#pragma GCC diagnostic ignored` and `#pragma push` and `pop`, but those pragma declarations seem to be ignored with the LTO build. I have been searching and it seems this happens since the compiler is invoked back at link stage in these cases.

If we finally decide to disable the warning, is there any other approach to target these warnings locally?

Thank you!